src/agent.cpp

```diff
@@ -82,6 +83,14 @@ namespace Rep
 
     std::string Agent::escape(const std::string& query)
     {
-        return Url::Url(query).defrag().escape().fullpath();
+      	std::regex escaped_slash ("%2[Ff]");
+      	std::regex escaped_newline ("\n");
```

Comment:
Using a newline character for this seems like a gross hack rather than a long term solution. I'd really prefer a fix that was cleaner.